xonsh/shell.py

```diff
@@ -11,6 +11,20 @@
 from xonsh.platform import (best_shell_type, has_prompt_toolkit,
                             ptk_version_is_supported)
 from xonsh.tools import XonshError, to_bool_or_int
+from xonsh.events import events
+
+
+events.doc('on_precommand', """
```

Comment:
I think we should keep it. None of the existing events grammatically flow; they all read like "On event <opaque identifier>" and not like sentences.